### PR TITLE
[LA-286] Improved error handling for s3 multi-part upload and canvas file download interactions.

### PR DESCRIPTION
**Pull request by sandeepmjay**
Improved error handling while interacting with canvas data api and working with s3 multi part uploads. The code records errors and degrades gracefully

### CODE REVIEW

File: lib/store/storage.js [hunk 1]
```diff
@@ -112,27 +112,42 @@ var storeExtractsOnS3 = module.exports.storeExtractsOnS3 = function(file, callba
...
-          log.info({file: file.filename, duration: end - start}, 'Finished Canvas data file upload');
-        });
-    } else if (res.statusCode === 403) {
-      log.error('Authorization error. Check if Canvas data file url expired.');
-    } else {
-      log.error({code: res.statusCode, msg: 'Status not 200.'});
-    }
+  request(file.url)
+    .on('response', function(res) {
+      // log.info(res.statusCode);
```
Comment (johncrossman) [inline lib/store/storage.js]: Minor: delete line above.

**Comment by johncrossman** [comment]
@sandeepmjay I'm done w/ review. Does the `bunyan` question cause any concern?

File: lib/store/storage.js [hunk 2]
```diff
@@ -112,27 +112,42 @@ var storeExtractsOnS3 = module.exports.storeExtractsOnS3 = function(file, callba
...
+      if (res.statusCode === 200) {
+        res.pipe(s3upload)
+          .on('error', function(err) {
+            log.error(err);
+            log.error({err: err, file: file.filename}, 'Unable to gunzip Canvas data file stream');
```
[the remark this answers is not in the document]
Comment (paulkerschen) [reply lib/store/storage.js]: @johncrossman @sandeepmjay If data-loch is using bunyan-prettystream, then yes, I think the error may fail to show up as per the SuiteC ticket. It's a pretty common pattern in SuiteC to log `err.message` specifically.

[the remark this answers is not in the document]
Comment (johncrossman) [reply lib/store/storage.js]: Yes, data-loch is using `bunyan-prettystream`.  @sandeepmjay I recommend we amend.

[the remark this answers is not in the document]
Comment (sandeepmjay) [reply lib/store/storage.js]: It does @johncrossman. From the log

Finished Canvas data file upload (file=group_membership_dim-00000-db9f076f.gz, duration=3663)

**Comment by johncrossman** [comment]
@sandeepmjay You'll find [odd behavior, in certain conditions, documented w/ bunyan](https://github.com/trentm/node-bunyan/issues/369).  But we'll see... Merging.

**Comment by sandeepmjay** [comment]
Thanks @johncrossman  @paulkerschen for the review.

**Pull request merged**